corehq/apps/users/views/mobile/users.py

```diff
@@ -772,6 +772,11 @@ def _build_commcare_user(self):
             self.new_mobile_worker_form.cleaned_data['force_account_confirmation']
             or self.new_mobile_worker_form.cleaned_data['force_account_confirmation_by_sms'])
 
+        role_id = None
+        default_cc_user_role = UserRole.commcare_user_default(self.domain)
+        if default_cc_user_role:
```

Comment:
The code is a little inconsistent in how a missing default role is handled.  It's an error, and we notify exception, but I can see not wanting to block critical workflows given that the fallback behavior is the same.  `user_role_choices` by contrast will fail, which I'm guessing would cause a 500 on that page.  Could you maybe explain a little about how you're thinking of that edge case?

Reply:
I was conflicted about this. I think the error handling I added in `commcare_user_default` was mostly for tests which fail if the role doesn't exist. It's probably best to let this 500 if the role is missing: 7bd7683cfc5